scripts/migrate.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# This command is used to automate the store, init, migrate process of
+# smart contracts.
+# Examples:
+
+# `sh scripts/migrate.sh ./artifacts/financial_snapshot_contract.wasm`
+# `sh scripts/migrate.sh ./artifacts/account_history_contract.wasm '{"limit": 10, "value_denom": "uusdc", "expiration": {"at_time": "604800000000000"}}'`
+
+migrate_contract() {
+    local contract_wasm_path="$1"
+    # Optional object parameter with a default value of '{}'
+    local instantiate_msg="$2"
+
+    # Check if $2 is provided, otherwise initialize to '{}'
+    if [ -z "$instantiate_msg" ]; then
+        instantiate_msg='{}'
+    fi
+
+    # Tries to store the contract and confirms the transaction
+    echo "Storing the contract..."
+    echo "y" | elysd tx wasm store "$1" --from cw --keyring-backend test --chain-id elystestnet-1 --gas auto --gas-adjustment=1.3 --fees 100000uelys -b sync
+
+    # Wait for a few seconds to allow for synchronization
+    sleep 1
+
+    # Parse the JSON output to extract the latest code information
+    codes=$(elysd q wasm list-code)
+    code_id=$(echo "$codes" | awk -F'code_id: "|"' '/code_id/{latest=$2} END{print latest}')
+
+    # Wait for a few seconds before initiating the contract
+    sleep 1
+
+    # Init the contract
+    echo "Initializing the contract..."
+    elysd tx wasm init $code_id "$instantiate_msg" --from cw --keyring-backend test --chain-id elystestnet-1 --gas auto --gas-adjustment=1.3 --fees 100000uelys -b sync -y --admin cw --label contract
+
+    # Wait for a few seconds before initiating the contract
+    sleep 1
+
+    # Get contract address based on code ID
+    contract_address=$(elysd q wasm contracts $code_id | awk '/^contracts:/{getline; print $2}')
+
+    # Migrate the contract
+    echo "Migrating the contract..."
+    elysd tx wasm migrate $contract_address $code_id '{}' --from cw --keyring-backend test --chain-id elystestnet-1 --gas auto --gas-adjustment=1.3 --fees 100000uelys -b sync -y
```

Comment:
@Wgil thanks so much for preparing this script, is there a reason why both init and migrate are called one after the other, init is supposed to be call once for contract initialization stage and then whenever there is a new version that we want the contract to use we use migrate

Reply:
@cosmic-vagabond sorry I was confused thinking that we needed to execute the three commands for local dev but certainly there is no need for that. I just talked to @kenta-elys and he only uses `store` and `init` in his local workflow.

Reply:
Just updated it and added an option for extra common params like node, etc.